website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,547 @@
+
+To do so, you can create a new file called `generate_schema_name.sql` under your macros folder with the following code:
+
+
+<File name='macros/generate_schema_name.sql'>
+
+
+```jinja
+{% macro generate_schema_name(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if custom_schema_name is none -%}
+
+        {{ default_schema }}
+
+    {%- elif  env_var('DBT_ENV_TYPE','DEV') == 'PROD' -%}
+        
+        {{ custom_schema_name | trim }}
+
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+
+```
+</File>
+
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Target database| Target schema | Resulting object               |
+|-------------|:-------------:|:-------------:|:------------------------------:|
+| Developer 1 | dev           | dbt_dev1      |dev.dbt_dev1_marketing.my_model  |
+| Developer 2 | dev           | dbt_dev2      |dev.dbt_dev2_marketing.my_model  |
+| CI PR 123   | ci            | dbt_pr_123    |ci.dbt_pr_123_marketing.my_model|
+| CI PR 234   | ci            | dbt_pr_234    |ci.dbt_pr_234_marketing.my_model|
+| Production  | prod          | analytics     |prod.marketing.my_model         |
+
+
+:::note
+
+We added logic to check if the current dbt run is happening in production or not. This is important, and we explain why in the [What not to do](/guides/customize-schema-alias?step=3#what-not-to-do) section.
+
+:::
+
+
+### 2. Static schemas: Add developer identities to tables
+
+Occasionally, we run into instances where the security posture of the organization prevents developers from creating schemas and all developers have to develop in a single schema.
+
+In this case, we can: 
+
+- Create a new file called generate_schema_name.sql under your macros folder with the following code:
+
+- Change `generate_schema_name()` to use a single schema for all developers, even if a custom schema is set.
+- Update `generate_alias_name()` to append the developer alias and the custom schema to the front of the table name in the dev environment.
+    - This method is not ideal, as it can cause long table names, but it will let developers see in which schema the model will be created in production.
+
+<File name='macros/custom_schema_name.sql'>
+
+```jinja
+
+{% macro generate_schema_name(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if custom_schema_name is none -%}
+
+        {{ default_schema }}
+
+    {%- elif  env_var('DBT_ENV_TYPE','DEV') != 'CI' -%}
+        
+        {{ custom_schema_name | trim }}
+
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+</File>
+
+
+<File name='macros/generate_schema_name.sql'>
```

Comment:
since this is the alias macro, shouldn't this be 'macros/generate_alias_name' and not 'generate_schema_name.sql'?

Reply:
Yep! You're right again!